fix cell background color on default and selected state

LaserPatrick opened the pull request.

rrolland commented on Segmentio/Source/Cells/SegmentioCell.swift:
```diff
@@ -139,6 +139,8 @@ class SegmentioCell: UICollectionViewCell {
             segmentTitleLabel?.textColor = selected ? selectedState.titleTextColor : defaultState.titleTextColor
             segmentTitleLabel?.font = selected ? selectedState.titleFont : defaultState.titleFont
+            backgroundColor = selected ? selectedState.backgroundColor : defaultState.backgroundColor
+            segmentImageView?.tintColor = selected ? selectedState.imageTintColor : defaultState.imageTintColor
```
@LaserPatrick In order to also manage style .onlyImage, you can move this code after the if statement.

LaserPatrick commented:
edited

Tykhonkov commented:
@LaserPatrick  can you provide more information about PR, and what issue it's suppose to fix?

LaserPatrick commented:
Hi @Tykhonkov Sure, my initial PR was due to the selectedState.backgroundColor was never used and then I applied it to the cell backgroundColor. Second I added the imageTintColor status to set the tint of the image if you use the templated image.

Finaly I removed the Void param to support swift 4.0 on xcode 9.0

Tykhonkov commented:
I have closed it because the color of cell shouldn't be changed.
This screen helps to understand why. 
![Preview](http://dl4.joxi.net/drive/2017/09/02/0024/3244/1608876/76/c805fe2181.jpg)

for changing background color you should use ```SegmentioOptions```

![Preview](http://joxi.net/EA4dEbjiDx51YA.jpg)

Image tint color is a good thing, but it should be changed only when animation of selection state is completed.

Swift 4 migration had been already done in another PR.
